python/dask_cudf/dask_cudf/io/parquet.py

```diff
@@ -177,65 +177,99 @@ def read_partition(
+                    and row_group is not None
+                    else row_group
                 )
-                paths = rgs = []
-                last_partition_keys = None
-            paths.append(path)
-            rgs.append(
-                [row_group]
-                if not isinstance(row_group, list) and row_group is not None
-                else row_group
-            )
-            last_partition_keys = partition_keys
+                last_partition_keys = partition_keys
 
-        dfs.append(
-            cls._read_paths(
-                paths,
-                fs,
-                columns=read_columns,
-                row_groups=rgs if rgs else None,
-                strings_to_categorical=strings_to_cats,
-                partitions=partitions,
-                partitioning=partitioning,
-                partition_keys=last_partition_keys,
-                **read_kwargs,
+            dfs.append(
+                cls._read_paths(
+                    paths,
+                    fs,
+                    columns=read_columns,
+                    row_groups=rgs if rgs else None,
+                    strings_to_categorical=strings_to_cats,
+                    partitions=partitions,
+                    partitioning=partitioning,
+                    partition_keys=last_partition_keys,
+                    **read_kwargs,
+                )
             )
-        )
-        df = cudf.concat(dfs) if len(dfs) > 1 else dfs[0]
+            df = cudf.concat(dfs) if len(dfs) > 1 else dfs[0]
 
-        # Re-set "object" dtypes align with pa schema
-        set_object_dtypes_from_pa_schema(df, schema)
-
-        if index and (index[0] in df.columns):
-            df = df.set_index(index[0])
-        elif index is False and df.index.names != (None,):
-            # If index=False, we shouldn't have a named index
-            df.reset_index(inplace=True)
+            # Re-set "object" dtypes align with pa schema
+            set_object_dtypes_from_pa_schema(df, schema)
+
+            if index and (index[0] in df.columns):
+                df = df.set_index(index[0])
+            elif index is False and df.index.names != (None,):
+                # If index=False, we shouldn't have a named index
+                df.reset_index(inplace=True)
+
+        except MemoryError as err:
+            raise MemoryError(
+                "Parquet data was larger than the available GPU memory!\n\n"
+                "Please try `split_row_groups=True` or set this option "
+                "to a smaller integer (if applicable).\n\n"
+                "If you are using dask-cuda workers, this may indicate "
+                "that the current `device_memory_limit` is too high. "
+                "If you are not using dask-cuda workers, this may indicate "
+                "that your workflow requires dask-cuda spilling.\n\n"
+                "Original Error: " + str(err)
+            )
+            raise err
```

Comment:
@VibhuJawa @ayushdg @randerzander - Ho do you feel about raising this "descriptive" error whenever we hit OOM while creating a new partition in `read_parquet`?

Reply:
Looks good but we might need to specify what setting it to an integer does more explicitly.  Like making sure that users understand setting `split_row_groups=x` will mean each partition will have x rowgroups.

Reply:
Looks good to me too. In addition to what Vibhu mentioned, my personal preference is to use language like setting `split_row_groups` "may" help or `device_memory_limit` "might" be too high. My main motivation for this is that the error might come up downstream when the user is doing a compute and some other task in the graph may be the culprit of high memory usage and the setting these values may not help in that case.

Reply:
Okay - I may justsimplify this to:

```python
        except MemoryError as err:
            raise MemoryError(
                "Parquet data was larger than the available GPU memory!\n\n"
                "See the `split_row_groups` argument in the `read_parquet` "
                "documentation.\n\n"
                "Original Error: " + str(err)
            )
            raise err
```

...and we can always expand/revise the message later.